### PR TITLE
Fix the interruption of the sending process due to a short network fa…

### PR DESCRIPTION
**Pull request by Nomidia**
…ilure

1. Network is unreachable  occasionally but the situation immediately resumes.
2. Save errno to prevent modification such as after calling strerror.

*Issue #, if available:*
#618 

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/source/Ice/SocketConnection.c [hunk 1]
```diff
@@ -329,7 +330,8 @@ STATUS socketSendDataWithRetry(PSocketConnection pSocketConnection, PBYTE buf, U
     while (socketWriteAttempt < MAX_SOCKET_WRITE_RETRY && bytesWritten < bufLen) {
         result = sendto(pSocketConnection->localSocket, buf, bufLen, NO_SIGNAL, destAddr, addrLen);
         if (result < 0) {
-            if (errno == EAGAIN || errno == EWOULDBLOCK) {
+            errorNum = errno;
```
Comment (MushMal) [inline src/source/Ice/SocketConnection.c:333]: I believe most implementations would store this value in TLS which is pretty fast but I like this change

**Comment by disa6302** [comment]
@Nomidia , 

Changes look good! Clang format check seems to be failing. Can you run `./clang-format.sh -f <filename>` and then run `./check-clang.sh` to confirming the formatting is right?

**Comment by disa6302** [comment]
@Nomidia , 

Something really funky happened when I tried to clang format and push the change to your branch. Can you see if you can reopen this PR?

**Comment by sharadrj** [comment]
Re-opening this pull request.

**Pull request merged**